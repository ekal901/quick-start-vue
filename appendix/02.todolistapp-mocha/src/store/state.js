// src/store/state.js
export default {
  todolist: [
    { id: 1, todo: "Watching Movie", done: false },
    { id: 2, todo: "Walking", done: true },
    { id: 3, todo: "ES6 Study", done: false },
    { id: 4, todo: "Basketball Game", done: false }
  ]
}