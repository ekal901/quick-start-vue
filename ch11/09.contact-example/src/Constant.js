// src/Constant.js
export default {
  // USE BOTH MUTATIONS AND ACTIONS
  ADD_CONTACT_FORM: "addContactForm",
  CANCEL_FORM: "cancelForm",
  EDIT_CONTACT_FORM: "editContactForm",
  EDIT_PHOTO_FORM: "editPhotoForm",
  FETCH_CONTACTS: "fetchContacts",

  // USE ONLY ACTIONS
  ADD_CONTACT: "addContact",
  UPDATE_CONTACT: "updateContact",
  UPDATE_PHOTO: "updatePhoto",
  DELETE_CONTACT: "deleteContact"
}