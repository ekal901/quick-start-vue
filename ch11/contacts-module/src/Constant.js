// src/Constant.js
export default {
  ADD_KEYWORD: 'addKeyword',
  SEARCH_CONTACT: "searchContact",
  BASE_URL: "http://sample.bmaster.kro.kr/contacts_long/search/"
}