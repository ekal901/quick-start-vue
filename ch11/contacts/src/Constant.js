// src/Constant.js
export default {
  SEARCH_CONTACT: "searchContact",
  BASE_URL: "http://sample.bmaster.kro.kr/contacts_long/search/"
}