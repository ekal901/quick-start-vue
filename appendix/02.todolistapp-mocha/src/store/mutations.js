// src/store/mutations.js
import Constant from '../Constant'
export default {
  [Constant.ADD_TODO]: (state, payload) => {
    if (payload.todo !== "") {
      state.todolist.push({ id: new Date().getTime(), todo: payload.todo })
    }
  },
  [Constant.DONE_TOGGLE]: (state, payload) => {
    var index = state.todolist.findIndex(item => item.id === payload.id);
    state.todolist[index].done = !state.todolist[index].done;
  },
  [Constant.DELETE_TODO]: (state, payload) => {
    var index = state.todolist.findIndex(item => item.id === payload.id);
    state.todolist.splice(index, 1);
  }
}