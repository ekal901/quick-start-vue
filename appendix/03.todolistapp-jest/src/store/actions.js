// src/store/actions.js
import Constant from '../Constant'
export default {
  [Constant.ADD_TODO]: (store, payload) => {
    store.commit(Constant.ADD_TODO, payload);
  },
  [Constant.DONE_TOGGLE]: (store, payload) => {
    store.commit(Constant.DONE_TOGGLE, payload);
  },
  [Constant.DELETE_TODO]: (store, payload) => {
    store.commit(Constant.DELETE_TODO, payload);
  }
}