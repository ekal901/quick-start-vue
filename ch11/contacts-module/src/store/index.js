// src/store/index.js
import Vue from 'vue';
import Vuex from 'vuex';
import Constant from '../Constant'
import module1 from './module1'

Vue.use(Vuex);
const store = new Vuex.Store({
  state: {
    keywordlist: []
  },
  mutations: {
    [Constant.ADD_KEYWORD]: (state, payload) => {
      state.keywordlist.splice(0, 0, payload.name)
    }
  },
  actions: {
    [Constant.ADD_KEYWORD]: (store, payload) => {
      store.commit(Constant.ADD_KEYWORD, payload)
    }
  },
  modules: { m1: module1 }
})

export default store;