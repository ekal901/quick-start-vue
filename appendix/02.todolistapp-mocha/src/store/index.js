// src/store/index.js
import Vue from 'vue'
import Vuex from 'vuex'
import state from './state'
import mutations from './mutations'
import actions from './actions'

Vue.use(Vuex); // Vuex를 전역에서 사용하도록 작성

const store = new Vuex.Store({
  state,
  mutations,
  actions
})

export default store;