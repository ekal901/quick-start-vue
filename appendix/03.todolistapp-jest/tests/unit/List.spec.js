// tests/unit/List.spec.js
import Vue from 'vue';
import Vuex from 'vuex';
import { mount, createLocalVue } from '@vue/test-utils';
import store from '@/store';
import List from '@/components/List';
import Constant from '@/Constant';

const localVue = createLocalVue();
localVue.use(Vuex);

describe("List.vue", () => {
  let wrapper;

  beforeAll(() => {
    wrapper = mount(List, { store, localVue });
    wrapper.vm.$store.dispatch(Constant.ADD_TODO, { todo: '스쿼시2' })
  })

  it('초기 렌더링 화면 테스트', () => {
    expect(wrapper.vm.$el.querySelectorAll('li').length).toEqual(5);
  })

  it('스냅샷 테스트(Jest)', (done) => {
    Vue.nextTick()
      .then(() => {

        expect(wrapper.vm.$el).toMatchSnapshot();
        done()
      })
      .catch(done)
  })
})