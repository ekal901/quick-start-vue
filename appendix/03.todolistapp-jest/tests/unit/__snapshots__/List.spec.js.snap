// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List.vue 스냅샷 테스트(Jest) 1`] = `
<span
  name="list"
  tag="ul"
>
  <li
    class=""
  >
    <span>
      Watching Movie
    </span>
     
    <!---->
     
    <span
      class="close"
    >
      ×
    </span>
  </li>
  <li
    class="checked"
  >
    <span>
      Walking
    </span>
     
    <span>
      (done)
    </span>
     
    <span
      class="close"
    >
      ×
    </span>
  </li>
  <li
    class=""
  >
    <span>
      ES6 Study
    </span>
     
    <!---->
     
    <span
      class="close"
    >
      ×
    </span>
  </li>
  <li
    class=""
  >
    <span>
      Basketball Game
    </span>
     
    <!---->
     
    <span
      class="close"
    >
      ×
    </span>
  </li>
  <li
    class=""
  >
    <span>
      스쿼시2
    </span>
     
    <!---->
     
    <span
      class="close"
    >
      ×
    </span>
  </li>
</span>
`;
